packages/transformers/js/src/JSTransformer.js

```diff
@@ -706,51 +706,64 @@ export default (new Transformer({
         asset.symbols.set(exported, local, convertLoc(loc));
       }
 
-      let deps = new Map(
-        asset
-          .getDependencies()
-          .map(dep => [dep.meta.placeholder ?? dep.specifier, dep]),
-      );
-      for (let dep of deps.values()) {
-        dep.symbols.ensure();
+      let deps = new Map();
+      for (let dep of asset.getDependencies()) {
+        if (!deps.has(dep.meta.placeholder ?? dep.specifier)) {
+          deps.set(dep.meta.placeholder ?? dep.specifier, []);
+        }
+        deps.get(dep.meta.placeholder ?? dep.specifier)?.push(dep);
+      }
+      for (let depArr of deps.values()) {
+        for (let dep of depArr) {
+          dep.symbols.ensure();
+        }
       }
 
       for (let {
         source,
         local,
         imported,
         loc,
+        kind,
       } of hoist_result.imported_symbols) {
-        let dep = deps.get(source);
-        if (!dep) continue;
-        dep.symbols.set(imported, local, convertLoc(loc));
+        for (let dep of nullthrows(deps.get(source))) {
+          if (
+            (dep.meta.kind === 'Require' ||
+              dep.meta.kind === 'Import' ||
+              dep.meta.kind === 'DynamicImport') &&
+            dep.meta.kind !== kind
+          ) {
+            continue;
+          }
+          dep.symbols.set(imported, local, convertLoc(loc));
+        }
```

Comment:
So there's a case where `dep.meta.kind` is `Export` and `kind` is `import` and I think we _should_ be setting the symbols in this case which is why I had the condition to continue if there's a mismatch _and_ if `dep.meta.kind` is an `ImportKind`

Reply:
So this would be the case where you have both reexports and imports of the same specifier in a single asset. But it looks like currently `import` always wins (so dep.meta.kind will be `import` in `asset.getDependencies()`). (Kind of strange anyway that we have `dep.meta.kind` anyway. AFAICT `dep.specifierType` would be sufficient for the packager, then there's only be esm or cjs)

And in this loop, `kind` is always `Import` or `Require`.

Reply:
Hmm in this test it looks like `Export` wins for `dep.meta.kind` so the symbols wouldn't get set if we have that single condition.
```
dep.sourcePath: /Users/bdo/parcel/packages/core/integration-tests/test/integration/scope-hoisting/es6/re-export-import/d.js
dep.meta.kind: Export
```
https://github.com/parcel-bundler/parcel/blob/6ab2b2ed5c9e5e224a203aa8073575498d3bd638/packages/core/integration-tests/test/scope-hoisting.js#L1148